components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -486,90 +486,360 @@ public User updateUser(User user, Map<String, Boolean> requiredAttributes) throw
+            String sortBy, String sortOrder, String domainName) throws CharonException {
 
         //Handle single attribute search
         if (node instanceof ExpressionNode) {
-            String attributeName = ((ExpressionNode) node).getAttributeValue();
-            String filterOperation = ((ExpressionNode) node).getOperation();
-            String attributeValue = ((ExpressionNode) node).getValue();
+            return getSingleAttributeFilteredUsers((ExpressionNode) node, requiredAttributes, offset, limit, sortBy,
+                    sortOrder, domainName);
 
-            if (log.isDebugEnabled()) {
-                log.debug(String.format("Listing users by filter: %s %s %s",
-                        attributeName, filterOperation, attributeValue));
-            }
+        } else if (node instanceof OperationNode) {
 
-            try {
-                if (isNotFilteringSupported(filterOperation)) {
-                    throw new NotImplementedException(String.format("System does not support filter operator: %s",
-                            filterOperation));
-                }
+            List<Object> filteredUsers = new ArrayList<>();
+            // 0th index is to store total number of results.
+            filteredUsers.add(0);
 
-                if (SCIMCommonUtils.isFilteringEnhancementsEnabled()) {
-                    if (SCIMCommonConstants.EQ.equalsIgnoreCase(filterOperation)) {
-                        if (StringUtils.equals(attributeName, SCIMConstants.UserSchemaConstants.USER_NAME_URI) &&
-                                !StringUtils.contains(attributeValue, CarbonConstants.DOMAIN_SEPARATOR)) {
-                            attributeValue = UserCoreConstants.PRIMARY_DEFAULT_DOMAIN_NAME +
-                                    CarbonConstants.DOMAIN_SEPARATOR + attributeValue;
-                        }
-                    }
+            // Support multi attribute filtering.
+            return getMultiAttributeFilteredUsers(node, requiredAttributes, offset, limit, sortBy, sortOrder,
+                    domainName, filteredUsers);
+        } else {
+            throw new CharonException("Unknown operation. Not either an expression node or an operation node.");
+        }
+    }
+
+    /**
+     * Method to filter users for a filter with a single attribute.
+     *
+     * @param node               Expression node for single attribute filtering
+     * @param requiredAttributes Required attributes for the response
+     * @param offset             Starting index of the count
+     * @param limit              Counting value
+     * @param sortBy             SortBy
+     * @param sortOrder          Sorting order
+     * @param domainName         Domain to run the filter
+     * @return User list with detailed attributes
+     * @throws CharonException Error while filtering
+     */
+    private List<Object> getSingleAttributeFilteredUsers(ExpressionNode node, Map<String, Boolean> requiredAttributes,
```

Comment:
Shall we reorder the name? Something similar to the ` getFilteredUsersBySingleAttribute`?. WDYT?

Reply:
Changed